output_tcp.go

```diff
@@ -88,8 +88,8 @@ func (o *TCPOutput) worker(bufferIndex int) {
 
 		if err != nil {
 			Debug(2, "INFO: TCP output connection closed, reconnecting")
-			o.buf[bufferIndex] <- msg
 			go o.worker(bufferIndex)
+			o.buf[bufferIndex] <- msg
```

Comment:
Overall it does makes sense, but I feel that it is temporary fix rather then permanent. Concurrency is still there. 

I wonder if better to replace `o.buf[bufferIndex] <- msg`, with the full switch statement with timeout protection, like:

```
    select {
    case o.buf[bufferIndex] <- msg:
    case <-time.After(10*time.Millisecond):
        fmt.Println("Channel is full, cannot send value")
    }
```

Reply:
I think this is a permanent fix. The proposed code can lead to a lost message. The changes will start a new go routine and the old go routine will block until the message is added back to the buffer.

Reply:
also in PluginWrite there is no such non blocking write to buffer

Reply:
I see, in this context it does makes sense.